src/settings-ui/Settings.UI/ViewModels/GeneralViewModel.cs

```diff
@@ -177,7 +195,304 @@ public GeneralViewModel(ISettingsRepository<GeneralSettings> settingsRepository,
+
+            var otherSoftwareText = "OS Build Version: " + GetOSVersion() + "\n.NET Version: " + GetDotNetVersion() + "\n\n";
+            var additionalInfo = HttpUtility.UrlEncode(otherSoftwareText);
+            var isElevatedRun = IsElevated ? "Running as admin: Yes" : "Running as admin: No";
+            var windowsSettings = ReportWindowsSettings();
+
+            var current_install_scope = GetCurrentInstallScope();
+
+            var installScope = current_install_scope == InstallScope.PerUser ? "Installation : User" : "Installation : System";
+
+            additionalInfo += windowsSettings + "%0a" + installScope + "%0a" + isElevatedRun;
+
+            var loadingMessage = new LoadingMessage();
+            loadingMessage.XamlRoot = App.GetSettingsWindow().Content.XamlRoot;
+
+            var cts = new CancellationTokenSource();
+
+            try
+            {
+                var showDialogTask = loadingMessage.ShowAsync().AsTask();
+                var bugReportTask = Task.Run(() => LaunchBugReport(cts.Token), cts.Token);
+
+                // Wait for either the dialog to be closed or the bug report to be generated
+                var completedTask = await Task.WhenAny(showDialogTask, bugReportTask);
+
+                if (completedTask == showDialogTask && showDialogTask.GetResultOrDefault() == ContentDialogResult.Primary)
+                {
+                    // Cancel the bug report task if the dialog was closed with Cancel
+                    await cts.CancelAsync();
+                }
+                else if (completedTask == bugReportTask)
+                {
+                    loadingMessage.Hide();
+
+                    // Bug report task completed
+                    string reportResult = await bugReportTask;
+
+                    gitHubURL = "https://github.com/microsoft/PowerToys/issues/new?assignees=&labels=Issue-Bug%2CNeeds-Triage&template=bug_report.yml" +
+                        "&version=" + version +
+                        "&additionalInfo=" + additionalInfo;
+
+                    var dialog = new ContentDialog
+                    {
+                        Title = string.IsNullOrEmpty(reportResult) ? string.Empty : reportResult,
+                        Content = string.IsNullOrEmpty(reportResult) ? "Failed to generate bug report." : "Bug report generated on your desktop. Please attach the file to the GitHub issue.",
+                        CloseButtonText = "OK",
+                        XamlRoot = loadingMessage.XamlRoot,
+                    };
+
+                    await dialog.ShowAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                await cts.CancelAsync();
+                loadingMessage.Hide();
+                var errorDialog = new ContentDialog
+                {
+                    Title = "Error",
+                    Content = $"An error occurred: {ex.Message}",
+                    CloseButtonText = "OK",
```

Comment:
Errors are not very _okay_ 🤓

Reply:
![error](https://github.com/user-attachments/assets/12fc1ff8-1f4c-4fd6-b7a7-e1d5fc06432b)

I manually throw an exception and error message looks like above. What do you think?